components/eamxx/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -94,8 +94,8 @@ void eam_pio_finalize() {
   eam_pio_finalize_c2f();
 }
 /* ----------------------------------------------------------------- */
-void register_file(const std::string& filename, const FileMode mode) {
-  register_file_c2f(filename.c_str(),mode);
+void register_file(const std::string& filename, const FileMode mode, const int iotype) {
```

Comment:
Up to you, but since you often call this with `iotype=IoType::DefaultIoType`, you could consider adding the default value in the fcn signature, so that you don't need to add `, scorpio::IOType::DefaultIOType` to all `register_file` calls..

Reply:
I think its a good idea to add the default value here (On the other hand forcing users to specify the args can prevent copy-paste errors :) ). I will make that change

Reply:
Yeah, it's not an obvious choice. As I said, up to you. If you think in the vast majority of cases the "default" value is fine, then adding defaults to the signature makes sense. But if the cost of using default when you shouldn't is very high, then we can just force users to always pass it.

Reply:
I will modify the code to use default args since there are many instances where the default io type is used (including tests etc)